<img src=http://www.acofi.edu.co/eiei2016/wp-content/uploads/2016/09/Logo-Universidad-ICESI.jpg width="300" height="200" align="center">

#**<big><big>Matemáticas Aplicadas III</big></big><br><small>Depto. Ciencias Físicas y Exactas de la FIDCA</small><br><small>_*Entrega 3*_ (Integrales triples y cambio de variables). </small>**

A continuación se le pide contestar o resolver por grupo las siguientes preguntas, utilizando una copia del presente cuaderno para consignar todas sus respuestas con el formato requerido.

La entrega final es el día **viernes 29 de agosto** y dos **grupos** estarán encargados de socializar sus resultados y avances el día **28 de agosto**.

**<u>Preguntas</u>**

A continuación encuentra separadas por celda cada pregunta, así usted solo debe continuar escribiendo en la celda para dar cada respuesta. No olvide que todas sus respuestas deben ir acompañadas de al menos un argumento geométrico o numérico y uno algebráico o en lenguaje natural.


**Nivel 3: Colaboración con IAG**

Pueden apoyarse en IAG para redactar, sintetizar o clarificar entregables, siempre y cuando
modifique activamente el contenido generado y lleve un registro de los prompts utilizados.
Ejemplo: Uso de IAG para estructurar un Cuaderno Colab que involucre simulación de optimización en
varias variables, explicando cómo se utilizó la herramienta y qué aportó al proceso


# 1. (6 ptos) **Integrales triples**

Para cada una de las siguientes integrales usted debe: i. Dibujar en un applet de GeoGebra la región de integración lo mas claro posible. ii. Hallar el valor de la integral numéricamente, es decir, utilizando la libreria scipy.

a. $$\int\int\int_W z\,dz\, dy\, dx,$$
Donde $W$ es la región acotada por los planos $x = 0, y = 0, z = 0, z = 1$ y el cilindro $x^2 + y^2 = 1$, con $x ≥ 0,\, y ≥ 0$.

b. $$\int\int\int_B x^2 \cos z \,dV.$$

Donde $B$ es la region acotada por $z = 0, z = π, y = 0, y = 1, x = 0$ y $x + y = 1$.

c. $$\int\int\int_S \,dx\, dy\, dz,$$
Donde $S=\{(x, y, z) : |x| ≤ 1, |y| ≤ 1, z ≥ 0 \mbox{ y }
x^2 + y^2 + z^2 ≤ 1\}$




##a.

Para establecer los límites de integración se debe tener en cuenta que $W$ está acotada por los planos $x = 0$, $y = 0$, $z = 0$, $z = 1$, y el cilindro $x^2 + y^2 = 1$, con $x ≥ 0$ y $y ≥ 0$. Eso nos deja únicamente la parte positiva del círculo en el plano $xy$. En altura, la región está limitada por los planos $z=0$ abajo y $z=1$ arriba.
<br>
Los límites de la variable $z$ son independientes a las variables $x$ y $y$. A su vez, sus límites están fijados por dos planos horizontales, $z=0$ y $z=1$, es decir, que los límites de integración de $z$ se escriben como:
$$0\le z\le 1$$
<br>
Ahora, los límites de integración de la variable $y$ partimos de la ecuación del cilindro $x^2 + y^2 = 1$.
<br>
Si despejamos $y$ obtenemos:
$$x^{2}+y^{2}=1$$
$$y^{2}=1-x^{2}$$
$$y=\sqrt{1-x^{2}}$$
Tomamos la raíz positiva porque la condición $y\ge 0$ nos deja solamente en el primer cuadrante.El límite inferior de $y$ es entonces el eje $x$, es decir $y=0$. Esto se debe a que $y$ no puede tomar valores negativos dentro de la región. Entonces, para cada valor fijo de $x$, $y$ comienza sobre el eje y puede crecer hasta la curva del cilindro. De manera que:
$$0\le y\le \sqrt{1-x^{2}}$$
Finalmente, los límites para $x$ se obtienen mirando la proyección de $x$ sobre el eje $x$. Como vemos, $x$ no puede ser negativo, por lo que el límite inferior es $x=0$ y el límite superior es donde el círculo sorta el eje $x$, osea, cuando $y=0$.
$$x^{2}+0=1$$
$$x=1$$
<br>
Por lo tanto,
$$0\le x\le 1$$
<br>
Ahora con todos los limites de integración, la integral se puede escribir como
$$$$
$$\int_{0}^{1}\int_{0}^{\sqrt{1-x^{2}}}\int_{0}^{1}\: z\:dzdydx$$
$$$$
Si resolvemos la integral de manera numerica obtenemos que:

In [ ]:
from scipy import integrate
import numpy as np

# Función a integrar
f = lambda z, y, x: z  # OJO: tplquad recibe variables en orden (z, y, x)

# Integración triple
resultado, error = integrate.tplquad(
    f,
    0, 1,  # x desde 0 hasta 1
    lambda x: 0, lambda x: np.sqrt(1 - x**2),  # y desde 0 hasta sqrt(1-x^2)
    lambda x, y: 0, lambda x, y: 1  # z desde 0 hasta 1
)

print("=== Resultado de la integral triple ===")
print("Valor aproximado:", resultado)
print("Error estimado:", error)

=== Resultado de la integral triple ===
Valor aproximado: 0.39269908169872403
Error estimado: 4.416955690089708e-11


Como se puede observar el resultado es $\frac{\pi}{8}$, lo que equivale aproximadamente a $0.39269908169872403$

In [ ]:
import IPython.display as IP
iframe = IP.IFrame(src="https://www.geogebra.org/calculator/ntsfjc6j",
  width="1000px",
  height="1000px")
IP.display_html(iframe)

##b.


La región $B$ esta delimitada por los planos $z=0$, $z=\pi$, $y=0$, $y=1$, $x=0$ y por la recta $x+y=1$. Esto significa que en el plano $xy$ la región es un triángulo, y verticalmente se extiende desde $z=0$ hasta $z=\pi$.
<br>
Para la variable $x$, la región está limitada por el plano $x=0$ y por la recta $x+y=1$. Si despejamos $x$ en términos de $y$, obtenemos:
$$$$
$$x+y=1$$
$$x=1-y$$
$$$$
Por lo tanto, para cada valor de $y$, $x$ varía desde el eje $x$ hasta la recta, es decir
$$$$
$$0\le x\le 1-y$$
$$$$
Para la variable $y$ la región se encuentra entre $y=0$ y $y=1$, por lo que los límites son
$$$$
$$0\le y\le 1$$
$$$$
Ahora en dirección vertical, $z$ está limitada por los planos horizontales $z=0$ y $z=\pi$. Estos límites no dependen de $x$ ni de $y$ por lo que se escriben directamente como
$$$$
$$0\le z\le \pi$$
$$$$
Con todos los límites hallados, la integral se puede escribir como
$$$$
$$\int_{0}^{\pi}\int_{0}^{1}\int_{0}^{1-y}\: x^{2}\: cosz\: dxdydz$$
$$$$
Al calcular la integral se obtiene:

In [ ]:
from scipy import integrate
import numpy as np

# Función a integrar
f = lambda x, y, z: x**2 * np.cos(z)

# Integración triple usando coordenadas cartesianas
resultado, error = integrate.tplquad(
    f,
    0, np.pi,  # z
    lambda z: 0, lambda z: 1,  # y
    lambda y, z: 0, lambda y, z: 1 - y  # x
)

print("Resultado de la integral:", resultado)
print("Error estimado:", error)

Resultado de la integral: 1.586419093909772
Error estimado: 3.600250483615147e-14


Es decir, que el resultado de la integral es aproximadamente $0$.

In [ ]:
import IPython.display as IP
iframe = IP.IFrame(src="https://www.geogebra.org/calculator/a8vx5krs",
  width="1000px",
  height="1000px")
IP.display_html(iframe)

##c.


La región $S$ está definida por $|x| ≤ 1, |y| ≤ 1, z ≥ 0 \mbox{ y }
x^2 + y^2 + z^2 ≤ 1$.  Esto significa que estamos trabajando con la mitad superior de una esfera de radio 1, limitada verticalmente por $z\ge 0$ y horizontalmente por $|x|\le 1$ y $|y|\le 1$.
<br>
Para la variable $z$, la región empieza en el plano $z=0$ y llega hasa la superficie de la esfera, cuyo punto más alto está en $z=1$. Por lo tanto:
$$$$
$$0\le z\le 1$$
$$$$
Para la variable $y$ observamos que para cada valor fijo de $z$, los valores posibles de $y$ deben de estar dentro de la esfera, es decir:
$$$$
$$y^{2}+z^{2}\le 1$$
$$$$
$$y^{2}\le 1-z^{2}$$
Por lo tanto los límites de $y$ son:
$$$$
$$-\sqrt{1-z^{2}}\le y\le \sqrt{1-z^{2}}$$
$$$$
Finalmente, para la variable $x$, los valores dependen de $y$ y de $z$, porque deben de mantenerse dentro de la esfera, es decir
$$$$
$$x^{2}+y^{2}+z^{2}\le 1$$
$$$$
$$x^{2}\le 1-y^{2}-z^{2}$$
$$$$
Por lo tanto, los limites de $x$ son
$$$$
$$-\sqrt{1-y^{2}-z^{2}}\le x\le \sqrt{1-y^{2}-z^{2}}$$
$$$$
Con los límites anteriores, la integral puede escribirse como
c
Esta integral representa el volumen de la mitad superior de la esfera de radio 1, al calcular la integral se obtiene:

In [ ]:
from scipy import integrate
import numpy as np

# Función a integrar
f = lambda x, y, z: 1  # Integral de 1

# Integración triple en coordenadas cartesianas
resultado, error = integrate.tplquad(
    f,
    0, 1,  # Límites de integración de z (de 0 a 1)
    lambda z: -np.sqrt(1 - z**2), lambda z: np.sqrt(1 - z**2),  # Límites de integración de y
    lambda y, z: -np.sqrt(1 - y**2 - z**2), lambda y, z: np.sqrt(1 - y**2 - z**2)  # Límites de integración de x
)

print("=== Resultado de la integral triple ===")
print("Simbólico aproximado:", resultado)
print("Error estimado:", error)

=== Resultado de la integral triple ===
Simbólico aproximado: 2.094395102393198
Error estimado: 2.0004660150618747e-09


Es decir, que el resultado de la integral es de $\frac{2\pi}{3}$, lo que es equivalente a aproximadamente $2.094395102393198$.

In [ ]:
import IPython.display as IP
iframe = IP.IFrame(src="https://www.geogebra.org/calculator/ppcy6sq6",
  width="1000px",
  height="1000px")
IP.display_html(iframe)

# 2. (24 ptos) **Geometría de funciones de $\mathbb{R^2}$ a $\mathbb{R^2}$**

a. Considere la transformación  $$ T(u,v)=\begin{bmatrix}1 & 1\\0 & 1\end{bmatrix}\begin{bmatrix}u\\v\end{bmatrix}.$$

Muestre que el determinante Jacobiano de $T$ es 1, y verifique, geométricamente, con un applet de GeoGebra que el triángulo, $D^*$, con vertices $(1, 0), (1, 1)$ y $(0, 1)$ y su imagen, $D=T(D^*)$, tienen la misma área. Explique con un argumento analítico por qué el área es igual en ambas regiones.

b. Sea $D$ el cuadrado con vértices $(0, 0), (1, 1), (2, 0), (1, −1)$ y $D^∗$ el paralelogramo con vértices $(0, 0), (1, 2), (2, 1), (1, −1)$. Encuentre una transformación lineal $T$, como la del punto anterior, que envíe $D^*$ en $D$.

c. Defina
$$T (x^∗, y^∗) = \begin{bmatrix}\frac{x^∗- y^∗}{\sqrt{2}}\\\frac{x^∗+ y^∗}{\sqrt{2}}\end{bmatrix}$$

Muestre que $T$ rota el cuadrado unitario, $D^∗ = [0, 1]×[0, 1]$.

OPCIONAL. Suponga que $T : \mathbb{R^2}→ \mathbb{R^2}$ es lineal y es dada por $T (x) =Ax$, donde $A$ es una matriz $2 × 2$ tal que $\det A \not= 0$. Muestre, utilizando un argumento geométrico y otro de tipo analítico, que si $P^∗$ es un paralelogramo entonces  $T (P^∗) = P$ también es un paralelogramo.






#$Solucion$ $2$

### $a).$

Sea
$$
T(u,v)=
\begin{bmatrix}1&1\\[2pt]0&1\end{bmatrix}
\begin{bmatrix}u\\ v\end{bmatrix}
=
\begin{bmatrix}u+v\\ v\end{bmatrix}
$$

Jacobiano:
$$
J_T(u,v)=
\begin{bmatrix}
\frac{\partial (u+v)}{\partial u} & \frac{\partial (u+v)}{\partial v}\\[4pt]
\frac{\partial v}{\partial u} & \frac{\partial v}{\partial v}
\end{bmatrix}
=
\begin{bmatrix}1&1\\0&1\end{bmatrix},
\qquad
\det J_T=1
$$

Imagenes de los vértices del triángulo $D^*$:
$$
(1,0)->(1,0),\quad
(1,1)->(2,1),\quad
(0,1)->(1,1)
$$

Area de $D^*$:
$$
\text{Área}(D^*)=\tfrac12 \big|\det\big((0,1),(-1,1)\big)\big|=\tfrac12
$$

Area de $D$:
$$
\text{Área}(D)=\tfrac12 \big|\det\big((1,1),(0,1)\big)\big|=\tfrac12
$$

Entonces como $\det J_T=1$, la transformacion preserva el area. Efectivamente, $D^*$ y $D$ tienen la misma area

---

##Comprobando con un Applet de geogebra

In [ ]:
from IPython.display import HTML

html_code = """
<iframe src="https://www.geogebra.org/classic/shfc6xp3?embed" width="800" height="600" allowfullscreen
style="border: 1px solid #e4e4e4;border-radius: 4px;" frameborder="0"></iframe>
"""

display(HTML(html_code))

Para una transformación lineal $T(x)=Ax$ y para cualquier paralelogramo generado por vectores $v,w$, el área del paralelogramo es
$$
\lvert \det[v \; w] \rvert.
$$
Aplicando $A$, los vectores se transforman en $Av,Aw$, y el area se convierte en
$$
\lvert \det[Av \; Aw] \rvert = \lvert \det A \rvert \, \lvert \det[v \; w] \rvert.
$$
De esta forma, el factor de escala de áreas bajo $T$ es precisamente $\lvert \det A \rvert$. En el caso particular en que $\lvert \det A \rvert = 1$, se concluye que el area de las regiones originales y sus imágenes es la misma.



### $b).$

Los vértices de $D$ son
$$
(0,0),\ (1,1),\ (2,0),\ (1,-1),
$$
con vectores generadores
$$
v'=(1,1),\quad w'=(1,-1).
$$

Los vértices de $D^*$ son
$$
(0,0),\ (1,2),\ (2,1),\ (1,-1),
$$
con generadores
$$
v=(1,2),\quad w=(1,-1).
$$

Buscamos $A$ tal que $Av=v'$ y $Aw=w'$.

Definimos
$$
V=\begin{bmatrix}1&1\\2&-1\end{bmatrix},\qquad
V'=\begin{bmatrix}1&1\\1&-1\end{bmatrix}.
$$

Entonces
$$
A = V'V^{-1}
= \begin{bmatrix}1&0\\[4pt]-\tfrac13&\tfrac23\end{bmatrix}.
$$

Verificación:
$$
A(1,2)=(1,1),\quad A(1,-1)=(1,-1),\quad A(2,1)=(2,0).
$$

Por tanto, $A(D^*)=D$.


In [ ]:
from IPython.display import HTML

html_code = """
<iframe src="https://www.geogebra.org/classic/sqe5zr6m?embed" width="800" height="600" allowfullscreen
style="border: 1px solid #e4e4e4;border-radius: 4px;" frameborder="0"></iframe>

"""

display(HTML(html_code))

### $c).$

Definimos
$$
T(x^*,y^*)=
\begin{bmatrix}
\dfrac{x^*-y^*}{\sqrt{2}}\\[6pt]
\dfrac{x^*+y^*}{\sqrt{2}}
\end{bmatrix},
\qquad
R=\frac{1}{\sqrt{2}}
\begin{bmatrix}1&-1\\1&1\end{bmatrix}.
$$

Propiedades de $R$:

- Sus columnas tienen norma $1$ y son ortogonales.  
- $\det R = 1$.  

Por lo tanto, $R$ es una rotación.

Comparando con la forma general de una rotacion en $\mathbb{R}^2$,
$$
\begin{bmatrix}
\cos \theta & -\sin \theta \\
\sin \theta & \cos \theta
\end{bmatrix},
$$
vemos que $\cos \theta = \sin \theta = 1/\sqrt{2}$, luego $\theta = 45^\circ$.  

Por lo tanto, $T$ rota el cuadrado unitario $45^\circ$ antihorariamente alrededor del origen.

Angulo de rotación:
$$
R =
\begin{bmatrix}
\cos 45^\circ & -\sin 45^\circ\\[4pt]
\sin 45^\circ & \cos 45^\circ
\end{bmatrix},
$$
luego $T$ es una rotación de $45^\circ$ antihoraria.

Imágenes de los vértices del cuadrado unitario $D^*=[0,1]\times[0,1]$:
\begin{align*}
(0,0)&\mapsto(0,0),\\
(1,0)&\mapsto\left(\tfrac{1}{\sqrt2},\tfrac{1}{\sqrt2}\right),\\
(0,1)&\mapsto\left(-\tfrac{1}{\sqrt2},\tfrac{1}{\sqrt2}\right),\\
(1,1)&\mapsto(0,\sqrt2).
\end{align*}

Estos puntos conforman el cuadrado unitario rotado $45^\circ$ al rededor del origen, por lo que $T$ rota el cuadrado unitario.

La matriz
$$
R = \frac{1}{\sqrt{2}}\begin{bmatrix}1 & -1 \\ 1 & 1\end{bmatrix}
$$
es una rotacion porque cumple las propiedades de una matriz de rotacion:

1. Sus columnas son ortonormales (norma 1 y ortogonales entre si), lo que garantiza que se preservan longitudes y angulos.
2. Su determinante es 1, lo que mantiene la orientacion y el area.


In [ ]:
from IPython.display import HTML

html_code = """
<iframe src="https://www.geogebra.org/classic/ta7xys7e?embed" width="800" height="600" allowfullscreen
style="border: 1px solid #e4e4e4;border-radius: 4px;" frameborder="0"></iframe>

"""

display(HTML(html_code))

### Conclusion

La transformacion $T$ definida por
$$
T(x^*,y^*)=
\begin{bmatrix}
\dfrac{x^*-y^*}{\sqrt{2}}\\[2pt]
\dfrac{x^*+y^*}{\sqrt{2}}
\end{bmatrix}
$$
es una rotacion lineal de 45 grados antihoraria alrededor del origen. Esto se deduce de varias propiedades analiticas: primero, las columnas de la matriz asociada $R$ son ortogonales y tienen norma 1, lo que garantiza que la transformacion no distorsiona longitudes ni angulos; segundo, el determinante de $R$ es 1, lo que indica que la orientacion y el area de las figuras se conservan.  

Al aplicar $T$ a los vertices del cuadrado unitario $D^*=[0,1]\times[0,1]$, se obtiene otro cuadrado con los mismos lados y angulos rectos, pero rotado 45 grados respecto a la posicion original. Esto confirma que $T$ no solo mantiene la forma y el area, sino que efectivamente realiza una rotacion pura del plano.  

En conclusion, $T$ es un ejemplo claro de una transformacion lineal que preserva la estructura geometrica de los objetos en $\mathbb{R}^2$, cambiando solo su orientacion mediante un giro de 45 grados alrededor del origen.


#3. (15 ptos)  **Cambio de variables**

i. Use coordenadas polares para evaluar la integral $$\int\int_D e^{−(x^2+y^2)}\, dA,$$
donde $D$ es el conjunto de puntos  $(x, y)$ tales que $1 ≤ x^2 +y^2 ≤ 4$.

ii. Explique por qué, si se desea evaluar la integral

$$\int\int_R (x + y)^2e^{x−y}\,dx dy,$$ donde $R$ es la región
acotada por $x + y = 1, x + y = 4, x − y = −1,$ y
$x − y = 1$, se debe dividir la región $R$ en mas de una subregión. Utilice un cambio de variable adecuado para transformar la región $R$ en una región $R^*$ que se deje evaluar con una sóla integral doble. Dibuje ambas regiones en un mismo applet de GeoGebra.

iii. Sea $R$ la región dentro de $x^2 + y^2 = 1$, pero fuera de
$x^2 + y^2 = 2y$ con $x ≥ 0, y ≥ 0$.

  (a) Dibuje en un applet de GeoGebra la región $R$.

  (b) Dibuje en un applet de GeoGebra la región $D$ definida en el plano uv  mediante el cambio de coordenadas $u = x^2 + y^2, v = x^2 + y^2 − 2y$.

  (c) Evalúe $$\int\int_R xe^y\;dx dy$$ usando el cambio de coordenadas $uv$, si y sólo si, con ese cambio de coordenadas se simplifica el cálculo de la integral.

##Solución 3.a

Teniendo en cuenta que $D$ es el conjunto de puntos $(x,y)$ donde $1 \leq x^2 + y^2 \leq 4$, y considerando que el dominio está expresado de forma que  $ D= {(x,y) \in \mathbb{R}, 1\leq x^2 + y^2 \leq 4} $. Donde $D$ describe una corona circular, la cual puede representarse bajo la norma euclideana de un vector en el que $||X|| = \sqrt{x^2 + y^2}$.

Por lo tanto para evaluar la integral
$$\int \int_{D} e^{-||X||^2} dA$$

Utilizamos la norma euclideana para convertir la integral a coordenadas polares, considerando la forma:
$$||X||^2 = x^2 + y^2$$
Donde:
- $x = r cos(\theta)$
- $y = r sin(\theta)$
- $||X||^2 = r^2$
- $dA = r \ dr \ d\theta$
- $\theta$ varía de $0$ a $2\pi$, para tener una cobertura completa en la que se recorre el círculo completamente alrededor del origen

$r^2 = ||X||^2$ ya que al sumar y elevar al cuadrado las coordenadas polares de $x$ y $y$ obtentemos $r^2$:
$$x^2+y^2=(r cos(\theta))^2+ (r sin(\theta)^2))$$ \
$$x^2+y^2=r^2(cos(\theta)^2+sin(\theta)^2)$$ \
$$x^2+y^2=r^2*1$$ \
$$x^2+y^2=r^2$$

Con lo anterior podemos describir $D$ en coordenadas polares de tal forma que $1 \leq r^2 \leq 4$, por lo que $1 \leq r \leq 2$

Reemplazando los datos obtenidos en la integral planteada
$$\int_{0}^{2\pi} \int_{1}^{2} e^{r^2} dr d\theta$$

Con esto evaluamos la integral interna utilizando sustitución, para el cual $u=r^2$, $du=2r dr → rdr=\frac{1}{2}du$. Para los límites, cuando $r=1$ entonces $u=1$ y cuando $r=2$ entonces $u=4$
Resolvemos la integral interna
$$\int_{1}^{4} e^{-u} du$$ \
$$\frac{1}{2} \int_{1}^{4} e^{-u} du$$ \
$$\frac{1}{2} \cdot -e^{-u}\Big|_{1}^{4}$$

Evaluamos los límites
$$\frac{1}{2}(-e^{-4}+e^{-1})$$ \
$$\frac{1}{2}(e^{-1}-e^{-4})$$

Con esto procedemos a resolver la integral externa
$$\int_{0}^{2\pi}\Big( \frac{1}{2}(e^{-1}-e^{-4}) \Big) d\theta$$
Al integrar y evaluar para $\theta$ obtenemos $2\pi$ por lo que el resultado de la integral queda como
$$2\pi \cdot \Big( \frac{1}{2}(e^{-1}-e^{-4}) \Big)$$

Por lo cual el resultado final de la integral es:
$$\pi(e^{-1}-e^{-4})$$ \
$$1.098187073$$



## Solución 3.b
La región $R$ está dada por ecuaciones de la forma $x+y=c_1$ y $x-y=c_2$, las cuales representan líneas paralelas con pendientes $-1$ y $1$, por lo que sus intersecciones generan los vértices que forman parte de la región $R$

Teniendo en cuenta la gráfica, los vértices que forman el paralelogramo en el plano $xy$ son:
* $(1,0)$ intersección de las rectas $(x+y=1)$ y $(x-y=1)$
* $(0,1)$ intersección de las rectas $(x+y=1)$ y $(x-y=1)$
* $(1.5, 2.5)$ intersección de las rectas $(x+y=4)$ y $(x-y=-1)$
* $(2.5, 1.5)$ intersección de las rectas $(x+y=4)$ y $(x-y=1)$

En la figura se observa que las rectas que actúan como borde izquierdo o derecho cambian según el valor de $y$, por lo que el paralelogramo tendría que partirse en subregiones ya que no se puede describir en todo su dominio con un solo intervalo de $y$ y un solo par de funciones para $x$, los bordes cambian de fórmula según el valor de $y$ o $x$ que se esté evaluando; el cambio de variables permite deshacernos de las fronteras diagonales para que sean rectas verticales u horizontales, de forma tal que se puede utilizar una única integral doblre para describir la región $R$ del rectángulo

Siguiendo con esta línea, el cambio de coordenadas queda como:
$$u = x+y$$\
$$v = x-y$$

Ahora despejamos para $y$ mediante eliminación
$$u = x+y$$\
$$- (v = x-y)$$ \
$$\rightarrow u-v=2y$$\
$$\rightarrow \frac{u-v}{2}=y$$

Y por sustitución obtenemos que $x$ es
$$u=x+\frac{u-v}{2}$$ \
$$u - \frac{u-v}{2} = x$$ \
$$\frac{2u-u+v}{2} = x$$ \
$$\frac{u+v}{2}=x$$


Una vez obtenidos $x$ y $y$ podemos calcular el jacobiano mediante la matriz jacobiana:



$$ J = \begin{vmatrix}
\frac{\partial x}{\partial u} & \frac{\partial x}{\partial v} \\
\frac{\partial y}{\partial u} & \frac{\partial y}{\partial v}
\end{vmatrix} $$

Reolvemos derivadas parciales para obtener los valores de la matriz:

$\frac{\partial x}{\partial u} = \frac{u+v}{2} du = \frac{1}{2}$ \
$\frac{\partial x}{\partial v} = \frac{u+v}{2} dv = \frac{1}{2}$ \
$\frac{\partial y}{\partial u} = \frac{u-v}{2} du = \frac{1}{2}$ \
$\frac{\partial y}{\partial v} = \frac{u-v}{2} dv = -\frac{1}{2}$

$$ J = \begin{vmatrix}
\frac{1}{2} & \frac{1}{2} \\
\frac{1}{2} & - \frac{1}{2}
\end{vmatrix} $$

Calculamos el jacobiano:

$J = \frac{\partial x}{\partial u} \cdot \frac{\partial y}{\partial v}
- \frac{\partial x}{\partial v} \cdot \frac{\partial y}{\partial u}$

$J = \frac{1}{2} \cdot -\frac{1}{2} - \frac{1}{2} \cdot \frac{1}{2}$

$J=-\frac{1}{2}$

Para el jacobiano tomamos su valor absoluto, por lo que su valor sería
$|J| = \frac{1}{2}$

Ahora bien, para transformar la región, consideramos que las combinaciones de las ecuaciones de la región nos permiten obtener las coordenadas: $(1, -1)$, $(1,1)$, $(4,-1)$, $(4,1)$, por lo cual tenemos que los límites de la región con las nuevas coordenadas se expresan como:
$$1 \leq u \leq 4$$
$$-1 \leq v \leq 1$$

Con esto ya tenemos que la región $R*$ es un rectángulo en el plano $uv$ que permite integrar de forma más sencilla

La integral transformada queda planteada como
$$\int_{-1}^{1} \int_{1}^{4} u^2e^v\frac{1}{2}  dudv$$

In [ ]:
from IPython.display import HTML

geogebra_url = "https://www.geogebra.org/calculator/w78azjur"

# Código HTML con el iframe de GeoGebra
html_code = f"""
<iframe src="{geogebra_url}" width="600" height="400" style="border:0px solid black;"></iframe>
"""

# Muestra el applet en el notebook
HTML(html_code)

### Solución 3.c

La región $R$ está delimitada por
* $x^2 + y^2 = 1$
* $x^2 + y^2 = 2y$ que se reescribe como $x^2 +(y-1)^1=1$

La figura representada corresponde a un círculo de radio 1 restringida al primer cuadrante $(x \geq 0, y\geq 0)$

Ahora definimos las variables para el plano $uv$
$$u=x^2+y^2$$ \
$$v=x^2+y^2-2y$$

Con el cambio podemos obtener las fronteras y restricciones adicionales
* $x^2+y^2=1$ se convierte en $u=1$
* $x^2+y^2=2y$ se convierte en $v=0$
* $x \geq 0$, $y \geq 0$ limita la región

Hallamos el jacobiano con las derivadas parciales
* $\frac{\partial u}{\partial x}=2x$
* $\frac{\partial u}{\partial y}=2y$
* $\frac{\partial v}{\partial x}=2x$
* $\frac{\partial v}{\partial y}=2y-2$

Contruimos la matriz jacobiana
$$ J = \begin{vmatrix}
2x & 2y \\
2x & 2y-2
\end{vmatrix} $$

Calculamos el determinante jacobiano
$$|J|=(2x)(2y-2)-(2y)(2x)$$ \
$$|J|=-4x$$

El jacobiano es entonces
$$dxdy=\frac{1}{|J|}dudv = \frac{1}{4x}dudv$$

En este caso el jacobiano depende de $x$, por lo cual resulta complejo reexpresar la integral en términos de $uv$

Ahora bien, como el cambio de variable $uv$ no simplifica el integrando, entonces se debe hacer por cordenadas polares, de forma tal que
$$x=rcos\theta$$ \
$$y=rsin\theta$$ \
$$dxdy = rdrd\theta$$

La integral original era
$$\int \int_R xe^ydxdy$$

Sin embargo, con el cambio a coordenadas polares se transformaria a
$$\int \int_R rcos\theta e^{r sin \theta} rdrd\theta$$

Esta solución no admite solución cerrada elemental por lo que resulta más viable realizarla numéricamente






In [ ]:
from IPython.display import HTML

geogebra_url = "https://www.geogebra.org/calculator/jjhfawmx"

# Código HTML con el iframe de GeoGebra
html_code = f"""
<iframe src="{geogebra_url}" width="600" height="400" style="border:0px solid black;"></iframe>
"""

# Muestra el applet en el notebook
HTML(html_code)

# 4. (5 puntos) **Coordenadas esféricas y cilíndricas**

Evalúe las siguientes integrales y dibuje la región de integración en un applet de Geogebra para cada una de ellas:

i. $$\int_0^3\int_0^{\sqrt{9-x^2}}\int_0^{\sqrt{9-x^2-y^2}} \frac{\sqrt{x^2 + y^2 + z^2}}{1 + [x^2 + y^2 + z^2]^2}\, dz dy dx,$$

ii. $$\int\int\int_W \frac{1}{\sqrt{x^2 + y^2 + z^2}}\;dx dy dz,$$ donde $W$ es
la región determinada por las condiciones $ \frac{1}{2}≤ z ≤ 1$ y $ x^2 + y^2 + z^2 ≤ 1$.

iii. Encuentre el valor promedio (sección 6.3 de Marsden $\&$ Tromba) de $ e^{−z}$ sobre la bola $
x^2 + y^2 + z^2 ≤ 1$.



# Solución
**i.**
$$
\int_0^3\int_0^{\sqrt{9-x^2}}\int_0^{\sqrt{9-x^2-y^2}}
\frac{\sqrt{x^2+y^2+z^2}}{1+\big(x^2+y^2+z^2\big)^2}\,dz\,dy\,dx
$$

$%%$

**Región.** Es la parte de la esfera $x^2+y^2+z^2\le 9$ en el primer octante ($x\ge 0,\ y\ge 0,\ z\ge 0$).

$%%$

**Cambio a esféricas:** $x=\rho\sin\phi\cos\theta,\ y=\rho\sin\phi\sin\theta,\ z=\rho\cos\phi,\ dV=\rho^2\sin\phi\,d\rho\,d\phi\,d\theta$.

$%%$

En el primer octante: $\theta\in[0,\tfrac{\pi}{2}],\ \phi\in[0,\tfrac{\pi}{2}]$, y por la esfera: $\rho\in[0,3]$.
Como $x^2+y^2+z^2=\rho^2$,

$%%$

$$
\frac{\sqrt{x^2+y^2+z^2}}{1+(x^2+y^2+z^2)^2}=\frac{\rho}{1+\rho^4}.
$$

**Cálculo:**
$$
\int_{0}^{\pi/2}\!\!\int_{0}^{\pi/2}\!\!\int_{0}^{3}
\frac{\rho}{1+\rho^4}\,\rho^2\sin\phi\,d\rho\,d\phi\,d\theta
$$

$%%$

$$
=\left(\int_0^3\frac{\rho^3}{1+\rho^4}\,d\rho\right)
\left(\int_0^{\pi/2}\sin\phi\,d\phi\right)
\left(\int_0^{\pi/2}d\theta\right).
$$

$%%$

Con $u=1+\rho^4$, $du=4\rho^3\,d\rho$:

$%%$

$$
\int_0^3\frac{\rho^3}{1+\rho^4}\,d\rho=\frac{1}{4}\ln(1+\rho^4)\Big|_0^3
=\frac{1}{4}\ln(82).
$$

$%%$

Además

$$\int_0^{\pi/2}\sin\phi\,d\phi=1 ∧ \int_0^{\pi/2}d\theta=\frac{\pi}{2}$$

**Resultado:**
$$
{\text{(i)}=\frac{\pi}{8}\,\ln(82).}
$$

In [ ]:
from IPython.display import HTML

geogebra_url = "https://www.geogebra.org/3d/njq3ukag"

# Código HTML con el iframe de GeoGebra
html_code = f"""
<iframe src="{geogebra_url}" width="1280" height="720" style="border:0px solid black;"></iframe>
"""

# Muestra el applet en el notebook
HTML(html_code)

**ii.**
$$
\iiint_{W}\frac{1}{\sqrt{x^{2}+y^{2}+z^{2}}}\,dx\,dy\,dz,
\quad
$$

$%%$

$$
W=\left\{(x,y,z):\tfrac{1}{2}\le z\le 1,\ x^{2}+y^{2}+z^{2}\le 1\right\}.
$$

$%%$

**Región:** Casquete de la esfera unitaria entre los planos $z=\tfrac12$ y $z=1$.

$%%$

**Cilíndricas:** $x=r\cos\theta,\ y=r\sin\theta,\ dV=r\,dr\,d\theta\,dz$.
De $x^2+y^2+z^2\le 1$ se obtiene $r\in\big[0,\sqrt{1-z^2}\big]$,
con $z\in[\tfrac12,1]$ y $\theta\in[0,2\pi]$.
El integrando es $1/\sqrt{r^2+z^2}$.

$%%$

**Cálculo:**
$$
\int_{z=1/2}^{1}\int_{\theta=0}^{2\pi}\int_{r=0}^{\sqrt{1-z^{2}}}
\frac{r}{\sqrt{r^{2}+z^{2}}}\,dr\,d\theta\,dz
=
\int_{1/2}^{1}\int_{0}^{2\pi}\left[\sqrt{r^{2}+z^{2}}\right]_{0}^{\sqrt{1-z^{2}}}\,d\theta\,dz
=
\int_{1/2}^{1}\int_{0}^{2\pi}(1-z)\,d\theta\,dz.
$$

$%%$

Así,
$$
2\pi\int_{1/2}^{1}(1-z)\,dz
=2\pi\left[z-\frac{z^{2}}{2}\right]_{1/2}^{1}
=2\pi\left(\frac12-\frac{3}{8}\right)
={\frac{\pi}{4}}
$$

In [ ]:
from IPython.display import HTML

geogebra_url = "https://www.geogebra.org/3d/nxsk2ctq"

# Código HTML con el iframe de GeoGebra
html_code = f"""
<iframe src="{geogebra_url}" width="1280" height="720" style="border:0px solid black;"></iframe>
"""

# Muestra el applet en el notebook
HTML(html_code)

**iii.**
El valor promedio es

$%%$

$$
\frac{1}{\operatorname{Vol}(B)}\iiint_{B} e^{-z}\,dV,
\quad \operatorname{Vol}(B)=\frac{4\pi}{3}.
$$

$%%$

En esféricas ($\rho\in[0,1],\ \phi\in[0,\pi],\ \theta\in[0,2\pi]$),
$z=\rho\cos\phi$ y $dV=\rho^{2}\sin\phi\,d\rho\,d\phi\,d\theta$:
$$

$%%$

$$
\iiint_{B} e^{-\rho\cos\phi}\,\rho^{2}\sin\phi\,d\rho\,d\phi\,d\theta
=2\pi\int_{0}^{1}\rho^{2}\left(\int_{0}^{\pi}e^{-\rho\cos\phi}\sin\phi\,d\phi\right)d\rho.
$$

$%%$

Con $u=\cos\phi$,

$%%$

$$
\int_{0}^{\pi}e^{-\rho\cos\phi}\sin\phi\,d\phi
=\int_{-1}^{1}e^{-\rho u}\,du
=\frac{e^{\rho}-e^{-\rho}}{\rho}
=\frac{2\sinh\rho}{\rho}.
$$

$%%$

Por tanto,

$%%$

$$
2\pi\int_{0}^{1}\rho^{2}\cdot\frac{2\sinh\rho}{\rho}\,d\rho
=4\pi\int_{0}^{1}\rho\sinh\rho\,d\rho
=4\pi\big(\rho\cosh\rho-\sinh\rho\big)\Big|_{0}^{1}
=4\pi\big(\cosh1-\sinh1\big)
=\frac{4\pi}{e}.
$$

Dividiendo entre el volumen:
$$
\text{Promedio}=\frac{(4\pi/e)}{(4\pi/3)}={\frac{3}{e}}
$$

In [ ]:
import mpmath as mp
mp.mp.dps = 50

I_rad = mp.quad(lambda rho: (rho**3)/(1+rho**4), [0, 3])
res_num = (mp.pi/2) * I_rad

res_ana = (mp.pi/8) * mp.log(82)

print("Integral (i) numérica =", res_num)
print("Integral (i) analítica =", res_ana)
print("Error absoluto (i)    =", abs(res_num - res_ana))


Integral (i) numérica = 1.7305146017047651432581278668806269922355018268395
Integral (i) analítica = 1.7305146017047651432581278668806269922355018268395
Error absoluto (i)    = 0.0


In [ ]:
import mpmath as mp
mp.mp.dps = 50

res_num = 2*mp.pi * mp.quad(lambda z: 1 - z, [0.5, 1.0])
res_ana = mp.pi/4

print("Integral (ii) numérica =", res_num)
print("Integral (ii) analítica =", res_ana)
print("Error absoluto (ii)     =", abs(res_num - res_ana))


Integral (ii) numérica = 0.78539816339744830961566084581987572104929234984378
Integral (ii) analítica = 0.78539816339744830961566084581987572104929234984378
Error absoluto (ii)     = 0.0


In [ ]:
import mpmath as mp
mp.mp.dps = 50

integral_total_num = 4*mp.pi * mp.quad(lambda r: r*mp.sinh(r), [0, 1])
vol_bola = 4*mp.pi/3
promedio_num = integral_total_num / vol_bola

integral_total_ana = 4*mp.pi / mp.e
promedio_ana = 3 / mp.e

print("Integral (iii) total numérica   =", integral_total_num)
print("Integral (iii) total analítica  =", integral_total_ana)
print("Error absoluto (integral total) =", abs(integral_total_num - integral_total_ana))
print()
print("Promedio (iii) numérico         =", promedio_num)
print("Promedio (iii) analítico        =", promedio_ana)
print("Error absoluto (promedio)       =", abs(promedio_num - promedio_ana))


Integral (iii) total numérica   = 4.6229093991636868716403727332507851964834040926577
Integral (iii) total analítica  = 4.6229093991636868716403727332507851964834040926577
Error absoluto (integral total) = 0.0

Promedio (iii) numérico         = 1.1036383235143269647865713104843826023374333930953
Promedio (iii) analítico        = 1.1036383235143269647865713104843826023374333930953
Error absoluto (promedio)       = 0.0


**La entrega solo se acepta en el formato de un cuaderno python colab .**